client/src/Components/Layouts/SingleCaptains/Popups/viewPopUp.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from "react";
+import { Modal, Table} from "antd";
+
+class View extends Component {
+  state = {
+    visible: this.props.visible,
+    id: this.props.id,
+    information: null
+  };
+
+  handleCancel = e => {
+    this.props.visibleFun("singleCaptain", "viewVisibility")(e);
+  };
+  componentWillReceiveProps(props) {
+    const { visible, id, information } = props;
+    this.setState({ visible, id, information });
```

Comment:
@engshorouq I think you could just use visible, id and information from props directly, without copying them into state.  That would make this component simpler (and would be fewer renders, since when componentWillReceiveProps copies the new props into state, it could cause another render).